test/test_c_interface.c

```diff
@@ -0,0 +1,152 @@
+{
+    open_f(BUFR_FILE_UNIT, INPUT_FILE);
+    openbf_f(BUFR_FILE_UNIT, "IN", BUFR_FILE_UNIT);
+
+    int iddate;
+    char msg_subset[SUBSET_STRING_LEN];
+    unsigned int subset_cnt = 0;
+
+    while (ireadmg_f(BUFR_FILE_UNIT, msg_subset, &iddate, SUBSET_STRING_LEN) == 0)
+    {
+        if (strcmp(subset, msg_subset))
+        {
+            while (ireadsb_f(BUFR_FILE_UNIT) == 0 && (subset_cnt < MAX_SUBSETS))
+            {
+                subset_cnt++;
+            }
+        }
+    }
+
+    closbf_f(BUFR_FILE_UNIT);
+    close_f(BUFR_FILE_UNIT);
+
+    return subset_cnt;
+}
+
+
+// Tests
+
+void test_readBufrFile()
+{
+    const char* subset = "NC021053";
+    const char* mnemonic = "CLAT";
+
+    int iret;
+    int iddate;
+    char msg_subset[SUBSET_STRING_LEN];
+    unsigned int subset_cnt = countSubsets(subset);
+
+    open_f(BUFR_FILE_UNIT, INPUT_FILE);
+    openbf_f(BUFR_FILE_UNIT, "IN", BUFR_FILE_UNIT);
+
+    double data_int = 0;
+    double data_rep = 0;
+    double* data_int_ptr = &data_int;
+    double* data_rep_ptr = &data_rep;
+    double data_int_buf[subset_cnt];
+    double data_rep_buf[subset_cnt];
+
+    int subset_idx = 0;
+    while (ireadmg_f(BUFR_FILE_UNIT, msg_subset, &iddate, SUBSET_STRING_LEN) == 0)
+    {
+        if (strcmp(subset, msg_subset))
+        {
+            while ((ireadsb_f(BUFR_FILE_UNIT) == 0) && (subset_idx < MAX_SUBSETS))
+            {
+                ufbint_f(BUFR_FILE_UNIT, (void**) &data_int_ptr, 1, 1, &iret, mnemonic);
+                ufbrep_f(BUFR_FILE_UNIT, (void**) &data_rep_ptr, 1, 1, &iret, mnemonic);
+                data_int_buf[subset_idx] = data_int;
+                data_rep_buf[subset_idx] = data_rep;
+                subset_idx++;
+            }
+        }
+    }
+
+    // Check a few values, and make sure they are good.
+    double check_vals[] = {45.818, 45.935, 46.046, 46.152};
+    for (unsigned int idx = 0; idx < 4; idx++)
+    {
+        if (fabs(data_int_buf[idx] - check_vals[idx]) > 0.001)
+        {
+            printf("%s", "Got bad data from BUFR file!");
+            exit(1);
+        }
+    }
+
+    // Check consistency between data read from ufbint and ufbrep
+    for (unsigned int idx = 0; idx < subset_cnt; idx++)
+    {
+        if (data_int_buf[idx] != data_rep_buf[idx])
+        {
+            printf("%s", "Data from ufbint didn't match ufbrep!");
+            exit(1);
+        }
+    }
+
+    closbf_f(BUFR_FILE_UNIT);
+    close_f(BUFR_FILE_UNIT);
+}
+
+
+void test_dxDump()
+{
+    open_f(BUFR_FILE_UNIT, INPUT_FILE);
+    open_f(DUMP_FILE_UNIT, TABLE_DUMP_FILE);
+    openbf_f(BUFR_FILE_UNIT, "IN", BUFR_FILE_UNIT);
+
+    dxdump_f(BUFR_FILE_UNIT, DUMP_FILE_UNIT);
+
+    closbf_f(BUFR_FILE_UNIT);
+    close_f(DUMP_FILE_UNIT);
+    close_f(BUFR_FILE_UNIT);
+}
+
+
+void test_mtInfo()
+{
+    open_f(BUFR_FILE_UNIT, INPUT_FILE);
+    openbf_f(BUFR_FILE_UNIT, "IN", BUFR_FILE_UNIT);
+
+    mtinfo_f("testfiles", TABLE_1_FILE_UNIT, TABLE_2_FILE_UNIT);
+
```

Comment:
How do you know mtinfo_f did what it was supposed to?  You aren't really using these master tables for anything in this test.

Reply:
Wondering if you have any Standard WMO BUFR files in the testfiles/data directory, otherwise I could add one Greg sent me.

Reply:
The tables subdirectory under the main project is chock-full of master table (i.e. Standard WMO BUFR) files that can be used with mtinfo_f.  This subdirectory already gets copied over to the build area for testing, so you can reference it there from any test code.  But the issue here is that mtinfo doesn't really do anything other than set the directory and unit numbers for reading master tables, which aren't actually done until a later call to ireadmg.  So until you call ireadmg, there's no way to know that it really found the tables, and which is why mtinfo can really only be tested within the scope of a larger program which is doing other things as well.  Again, test/test_IN_4.f shows a good example of this.

Reply:
@aerorahul the current test doesn't really ensure the file is present; again, all mtinfo does is just set the directory and Fortran logical unit numbers.

I just made a clone of this branch and will see if I can cook up a quick C test.  This would be good to have anyway.  But I'm presuming @rmclaren also has some C application he wants to use mtinfo with, and which is why he added this C binding to the bufr.interface.f90 file in the first place(?)

Reply:
then the test is to just set the directory and Fortran logical numbers.
It should do nothing more, because mtinfo does nothing more.

Reply:
I was going to add a new test readStdWmoBufrFile that calls mtinfo (similar to readNcepBufrFile). 

@jbathegit The function is being used in a c++ application but not in a form where you could just copy and paste the code into a test application...